src/functions/parse-date.js

```diff
@@ -2,25 +2,37 @@ import 'sugar';
 import moment from 'moment';
 import _ from 'lodash';
 
-export default (bot, string, base = moment(), separators = /\b(?:to|-|until|till)\b/i) => { // eslint-disable-line
+const SEPARATORS = /\b(?:to|-|until|till|for)\b/i;
+const KEYWORDS = /\b(?:from|since)\b/gi;
+export default (bot, string, base = moment(), separators = SEPARATORS) => { // eslint-disable-line
   moment.updateLocale('en', _.get(bot.config, 'moment') || {});
   moment.locale('en');
 
-  string = string.replace(/\b(?:from|since|for)\b/gi, '');
+  string = string.replace(KEYWORDS, '');
   if (separators.test(string)) {
+    const separator = separators.exec(string)[0];
     const [from, to] = string.split(separators);
+    const dates = {
+      from: moment(parse(from.trim())),
+      to: moment(parse(to.trim()))
+    };
 
-    const mfrom = moment(parse(from.trim()));
-    const mto = moment(parse(to.trim()));
+    if (separator === 'for') {
+      if (!dates.from.isValid()) {
```

Comment:
I think it can be good to replace this if with a ternary expression.
```js
dates.to = dates.from.isValid()
  ? dates.from.clone().add(dates.to.diff(moment()))
  : moment().add(dates.to.diff(moment()))
```